### PR TITLE
Init button

### PR DESCRIPTION
**Pull request by ianhi**
Closes: https://github.com/jupyterlab/jupyterlab-git/issues/589

New UI:
image

I think this still needs some work on where to put and how to style the file path. The two big issues are:
1. If you are in the jupyter root directory the file path looks weird:
image
I didn't see an easy way to get the absolute file path from the filebrowser in order to get the folder name in this case.

2. A really long folder name renders poorly
image

I briefly explored the suggestion of replacing the Current Repository header with the name of the current repo but didn't see a super easy way to do this. 

### Better styling ideas:

Maybe just don't list the name of the current folder. i.e. say: "You aren't currently in a git repository you can either ...."

Or: style the filepath text differently and limit its length.

### Slow refresh:
I also found that sometimes the filepath can be slow to update:
slow
slow2

### Adding a clone button
If https://github.com/jupyterlab/jupyterlab-git/pull/596 gets merged then it would be nice to add a third button for cloning to this interface.

### CODE REVIEW

**Comment by kgryte** [comment]
If I had to hazard a guess as to why things might be "slow" is because of the poll interval. Try adjusting the poll interval in the extension settings and seeing if you see the same lag.

**Comment by ianhi** [comment]
> " is because of the poll interval

Yup! that was definitely it. In the above gifs it updates every 3 seconds, which is the default polling rate.

I had a further thought about how to display the folder names.
1. For the jupyter root dir just have prepared text:
image

2. For folders over than the jupyter root level just display the final folder in the path in bold:
e.g. here Documents/van_kampen becomes just van_kampen
image

**Comment by fcollonval** [comment]
@ianhi Your latest proposal looks nice. You could even add a `title` on the folder name element to display the full path when hovered.

**Comment by ianhi** [comment]
Good suggestion with the title, I added that. I also made all three of the buttons be the same width because otherwise they were slightly misaligned.

**Comment by ianhi** [comment]
@fcollonval you had suggested removing the top part of the UI
image
when not in a git repo. I think this makes sense was a pretty easy fix. But, to me it makes the text feel almost too high up?
image

Also, looking at the test failures they don't seem to be related to these changes:
https://travis-ci.org/github/jupyterlab/jupyterlab-git/builds/677526123#L804
```bash
pluggy.manager.PluginValidationError: Plugin 'asyncio' could not be loaded: (pytest 4.3.1 (/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages), Requirement.parse('pytest>=5.4.0'))!

The command "pytest jupyterlab_git -r ap" exited with 1.
```

**Comment by ianhi** [comment]
Also this PR will make the gif in the readme slightly out of date.

**Comment by telamonian** [comment]
> Also this PR will make the gif in the readme slightly out of date.

Well, if you feel bad about it you can always record a new one :)

I used the free/oss app LICEcap to make the existing animation, if you actually do want to take a stab at it.

@ianhi I like the direction this PR is taking. I just pushed a fix, the tests are now passing

**Comment by fcollonval** [comment]
@ianhi Thanks for the great work. This looks quite good. I think the vertical space is find.

The only change, I would suggest, is to left align the text paragraph. This feels more natural for large text.

**Comment by ianhi** [comment]
I'll add the left align. I also need to resolve the conflicts, I think rebasing is the appropriate thing here?

I'd also like to add icons on the left side of the buttons (maybe inside the buttons?). Maybe a folder icon for the filebrowser, a plus sign for init, and the pull icon for clone. I think that after the first time you've seen the dialog and know what to expect the icons will make the options easier to visually process.

**Comment by fcollonval** [comment]
> I think rebasing is the appropriate thing here?

Yep - but be aware that the port to JupyterLab 2.0 has been merged. So this means you will need to install JLab 2.0 in your dev environment, reinstall the extension in it and rebuild the frontend.

> I'd also like to add icons on the left side of the buttons (maybe inside the buttons?)

Inside the button makes more sense.

**Comment by ianhi** [comment]
Rebased on master and squashed some of the commits together. New UI looks like:
image.

It feels as though I'm missing some punctuation in the second sentence, but proper comma placement has always been a weak point for me so I'm not quite sure.


I tried to add icons using `<LabIcon.resolveReact icon='git-pull' />` but wasn't able to scale the icon and keep the icon on the same line as the rest of the text in the button. For example:
sad
But I won't lose sleep over the lack of an icon.

File: src/components/GitPanel.tsx [hunk 1]
```diff
@@ -295,16 +302,42 @@ export class GitPanel extends React.Component<
...
-        <div>Unable to detect a Git repository.</div>
+        <div className={warningTextClass}>
+          {path ? (
+            <React.Fragment>
+              <b title={path}>{PathExt.basename(path)}</b> is not
+            </React.Fragment>
+          ) : (
+            'You are not currently in'
+          )}{' '}
+          a git repository. To use git navigate to a local repository, intialize
```
Comment (kgryte) [inline src/components/GitPanel.tsx:317]:
```suggestion
          a Git repository. To use Git, navigate to a local repository, initialize
```

**Comment by ianhi** [comment]
amended to fix the lint error. 

Unless anyone has a suggestion on how to properly incorporate icons inside the buttons then I'm happy with the way this looks.

**Comment by fcollonval** [comment]
@meeseeksdev backport to 0.11.x

File: src/components/GitPanel.tsx [hunk 2]
```diff
@@ -295,16 +302,42 @@ export class GitPanel extends React.Component<
...
             this.props.model.commands.execute('filebrowser:toggle-main')
           }
         >
-          Find a repository
+          Open the FileBrowser
+        </button>
+        <br />
+        <button
+          className={repoButtonClass}
+          onClick={() => this.props.model.commands.execute('git:init')}
```
[the remark this answers is not in the document]
Comment (ianhi) [reply src/components/GitPanel.tsx]: Thanks for leaving these comments. Very helpful as I learn more!

**Pull request merged**